roop/core.py

```diff
@@ -136,12 +136,14 @@ def preview_video(video_path):
 
 def select_face():
     args['source_img'] = filedialog.askopenfilename(title="Select a face")
-    preview_image(args['source_img'])
+    if args['source_img'] and args['source_img'] != "":
+        preview_image(args['source_img'])
```

Comment:
I suggest a small optimization which is to check that the file selected is not the same as the current one, to avoid reprocessing the preview:

```
source_img = filedialog.askopenfilename(title="Select a face")

if source_img and source_img != args.get('source_img'):
    args['source_img'] = source_img
    preview_image(args['source_img'])
```

Reply:
Good idea. That is even better!

Reply:
We still need the empty string check however. I pushed the new version now.